core/src/main/java/org/elasticsearch/action/search/SearchTransportService.java

```diff
@@ -76,10 +76,13 @@
     public static final String FETCH_ID_ACTION_NAME = "indices:data/read/search[phase/fetch/id]";
 
     private final TransportService transportService;
+    private final ResponseCollectorService responseCollectorService;
 
-    public SearchTransportService(Settings settings, TransportService transportService) {
+    public SearchTransportService(Settings settings, TransportService transportService,
+                                  ResponseCollectorService responseCollectorService) {
```

Comment:
you can also remove the dependency by passing a `BiFunction<Connection, SearchActionListener, ActionListener>` instead of ResponseCollectorService and require it to be non-null then it's easier to test? and we never need to do nullchecks